--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,9 @@
 # Makefile for the drm device driver.  This driver provides support for the
 # Direct Rendering Infrastructure (DRI) in XFree86 4.1.0 and higher.
 
-ccflags-y := -Iinclude/drm
+include /boot/config-$(KVER)
+
+ccflags-y := -I/var/lib/dkms/drm-dkms/3.10.0/build/include/drm
 
 drm-y       :=	drm_auth.o drm_buffer.o drm_bufs.o drm_cache.o \
 		drm_context.o drm_dma.o \
@@ -29,27 +31,33 @@ obj-$(CONFIG_DRM_KMS_HELPER) += drm_kms_
 CFLAGS_drm_trace_points.o := -I$(src)
 
 obj-$(CONFIG_DRM)	+= drm.o
-obj-$(CONFIG_DRM_USB)   += drm_usb.o
+#obj-$(CONFIG_DRM_USB)   += drm_usb.o
 obj-$(CONFIG_DRM_TTM)	+= ttm/
-obj-$(CONFIG_DRM_TDFX)	+= tdfx/
-obj-$(CONFIG_DRM_R128)	+= r128/
-obj-$(CONFIG_DRM_RADEON)+= radeon/
-obj-$(CONFIG_DRM_MGA)	+= mga/
-obj-$(CONFIG_DRM_I810)	+= i810/
+#obj-$(CONFIG_DRM_TDFX)	+= tdfx/
+#obj-$(CONFIG_DRM_R128)	+= r128/
+#obj-$(CONFIG_DRM_RADEON)+= radeon/
+#obj-$(CONFIG_DRM_MGA)	+= mga/
+#obj-$(CONFIG_DRM_I810)	+= i810/
 obj-$(CONFIG_DRM_I915)  += i915/
-obj-$(CONFIG_DRM_MGAG200) += mgag200/
-obj-$(CONFIG_DRM_CIRRUS_QEMU) += cirrus/
-obj-$(CONFIG_DRM_SIS)   += sis/
-obj-$(CONFIG_DRM_SAVAGE)+= savage/
-obj-$(CONFIG_DRM_VMWGFX)+= vmwgfx/
-obj-$(CONFIG_DRM_VIA)	+=via/
-obj-$(CONFIG_DRM_NOUVEAU) +=nouveau/
-obj-$(CONFIG_DRM_EXYNOS) +=exynos/
-obj-$(CONFIG_DRM_GMA500) += gma500/
-obj-$(CONFIG_DRM_UDL) += udl/
-obj-$(CONFIG_DRM_AST) += ast/
-obj-$(CONFIG_DRM_SHMOBILE) +=shmobile/
-obj-$(CONFIG_DRM_OMAP)	+= omapdrm/
-obj-$(CONFIG_DRM_TILCDC)	+= tilcdc/
-obj-$(CONFIG_DRM_QXL) += qxl/
+#obj-$(CONFIG_DRM_MGAG200) += mgag200/
+#obj-$(CONFIG_DRM_CIRRUS_QEMU) += cirrus/
+#obj-$(CONFIG_DRM_SIS)   += sis/
+#obj-$(CONFIG_DRM_SAVAGE)+= savage/
+#obj-$(CONFIG_DRM_VMWGFX)+= vmwgfx/
+#obj-$(CONFIG_DRM_VIA)	+=via/
+#obj-$(CONFIG_DRM_NOUVEAU) +=nouveau/
+#obj-$(CONFIG_DRM_EXYNOS) +=exynos/
+#obj-$(CONFIG_DRM_GMA500) += gma500/
+#obj-$(CONFIG_DRM_UDL) += udl/
+#obj-$(CONFIG_DRM_AST) += ast/
+#obj-$(CONFIG_DRM_SHMOBILE) +=shmobile/
+#obj-$(CONFIG_DRM_OMAP)	+= omapdrm/
+#obj-$(CONFIG_DRM_TILCDC)	+= tilcdc/
+#obj-$(CONFIG_DRM_QXL) += qxl/
 obj-y			+= i2c/
+
+all:
+	make -C /lib/modules/$(KVER)/build M=`pwd` modules
+
+clean:
+	make -C /lib/modules/$(KVER)/build M=`pwd` clean
--- a/i915/Makefile
+++ b/i915/Makefile
@@ -2,7 +2,7 @@
 # Makefile for the drm device driver.  This driver provides support for the
 # Direct Rendering Infrastructure (DRI) in XFree86 4.1.0 and higher.
 
-ccflags-y := -Iinclude/drm
+ccflags-y := -I/var/lib/dkms/drm-dkms/3.8/build/include/drm
 i915-y := i915_drv.o i915_dma.o i915_irq.o \
 	  i915_debugfs.o \
           i915_suspend.o \
