diff --git a/backports/backport-include/linux/idr.h b/backports/backport-include/linux/idr.h
index 737632b..8c604cf 100644
--- a/backports/backport-include/linux/idr.h
+++ b/backports/backport-include/linux/idr.h
@@ -38,7 +38,7 @@ static inline int idr_alloc(struct idr *idr, void *ptr, int start, int end,
 		if (!idr_pre_get(idr, gfp_mask))
 			return -ENOMEM;
 		ret = idr_get_new_above(idr, ptr, start, &id);
-		if (!ret && id > end) {
+		if (!ret && (end > 0 && id > end)) {
 			idr_remove(idr, id);
 			ret = -ENOSPC;
 		}
diff --git a/backports/drivers/gpu/drm/i915/intel_opregion.c b/backports/drivers/gpu/drm/i915/intel_opregion.c
index 119771f..895e090 100644
--- a/backports/drivers/gpu/drm/i915/intel_opregion.c
+++ b/backports/drivers/gpu/drm/i915/intel_opregion.c
@@ -293,11 +293,20 @@ static void intel_didl_outputs(struct drm_device *dev)
 	if (!handle || acpi_bus_get_device(handle, &acpi_dev))
 		return;
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,9,0))
 	if (acpi_is_video_device(handle))
+#else
+	if (acpi_is_video_device(acpi_dev))
+#endif
 		acpi_video_bus = acpi_dev;
 	else {
 		list_for_each_entry(acpi_cdev, &acpi_dev->children, node) {
-			if (acpi_is_video_device(acpi_cdev->handle)) {
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3,9,0))
+			if (acpi_is_video_device(acpi_cdev->handle))
+#else
+			if (acpi_is_video_device(acpi_cdev))
+#endif
+			{
 				acpi_video_bus = acpi_cdev;
 				break;
 			}
